pkg/utils/utils.go

```diff
@@ -64,3 +76,16 @@ func ParseSubjectReference(subRef string) (common.Reference, error) {
 func TrimSpaceAndToLower(input string) string {
 	return strings.ToLower(strings.TrimSpace(input))
 }
+
+// ParseRequestKey parses key string to a structured RequestKey object.
+func ParseRequestKey(key string) (RequestKey, error) {
+	re := regexp.MustCompile(subjectPattern)
+	match := re.FindStringSubmatch(key)
+	if match == nil || len(match) < 4 {
```

Comment:
suggestion to add a sample string " "docker.io/test/hello:v1" to explain why we expect "4"